sklearn/cluster/_agglomerative.py

```diff
@@ -506,7 +506,7 @@ def linkage_tree(
             # by sklearn.metrics.pairwise_distances.
             if X.shape[0] != X.shape[1]:
                 raise ValueError(
-                    "Distance matrix should be square, Got matrix of shape {X.shape}"
+                    f"Distance matrix should be square, Got matrix of shape {X.shape}"
```

Comment:
We can update the test:

https://github.com/scikit-learn/scikit-learn/blob/4b5ea125ed076004a3b071ef8d422fbdff89bcc5/sklearn/cluster/tests/test_hierarchical.py#L873-L874

to catch this error message:

```python
    with pytest.raises(
        ValueError,
        match=r"Distance matrix should be square, Got matrix of shape \(5, 3\)",
    ):
        AgglomerativeClustering(affinity="precomputed", linkage="complete").fit(X)
```

Reply:
Yes. I was referring to this one.